crates/hdx_parser/src/comparison.rs

```diff
@@ -13,25 +13,29 @@ pub enum Comparison {
 
 impl<'a> Parse<'a> for Comparison {
 	fn parse(parser: &mut Parser<'a>) -> Result<Comparison> {
-		Ok(match parser.next() {
-			Token::Delim('=') => Comparison::Equal,
-			Token::Delim('>') => {
-				if let Token::Delim('=') = parser.peek_with(Include::Whitespace) {
+		Ok(match (parser.next().kind(), parser.next().char()) {
+			(Kind::Delim, Some('=')) => Comparison::Equal,
+			(Kind::Delim, Some('>')) => {
```

Comment:
I wonder if it would be nicer as a macro, kind of like we have for `match_ignore_case`?